apps/cowswap-frontend/src/legacy/hooks/useFetchListCallback.ts

```diff
@@ -21,7 +30,11 @@ export function useFetchListCallback(): (listUrl: string, sendDispatch?: boolean
     async (listUrl: string, sendDispatch = true) => {
       const requestId = nanoid()
       sendDispatch && dispatch(fetchTokenList.pending({ requestId, url: listUrl, chainId }))
-      return getTokenList(listUrl, (ensName: string) => resolveENSContentHash(ensName, MAINNET_PROVIDER))
+      return getTokenList(listUrl, (ensName: string) => {
+        const cachedHash = TOKENS_LISTS_ENS_HASHES[ensName]
```

Comment:
But... then you are basically hardoding here the resolution

The hash of the name don't require calls:
https://github.com/cowprotocol/cowswap/blob/fc06ded36ddd082defec433b1ca7db77c2151f32/libs/common-utils/src/resolveENSContentHash.ts#L64

However, getting the resolver and the content hash needs two calls. 

If these list add a token, your solution will remain frozen, right?

Reply:
It requires:
```
const resolverAddress = await ensRegistrarContract.resolver(hash)
```

Reply:
>If these list add a token, your solution will remain frozen, right?

No, because it's a hash of the list name, not the content

Reply:
I'm not sure I follow you. `resolveENSContentHash` returns a hash, and you are taking a snapshot of the value of today. 

What I think `resolveENSContentHash` does is 3 things:
1. takes your `ensName` and convert it to a hash: (this requires no calls, and could be cacheable but also there's no big advantages in doing so): https://github.com/cowprotocol/cowswap/blob/fc06ded36ddd082defec433b1ca7db77c2151f32/libs/common-utils/src/resolveENSContentHash.ts#L64 

2. Get resolver: This requires a RPC call. A resolver in ENS can dynamically change by the owner of a domain, so this should not be cached ❌ https://github.com/cowprotocol/cowswap/blob/fc06ded36ddd082defec433b1ca7db77c2151f32/libs/common-utils/src/resolveENSContentHash.ts#L65 

3. Ask the resolver to resolve the hash from (1.) Also not cacheable, you can change the content of your list any time ❌ https://github.com/cowprotocol/cowswap/blob/fc06ded36ddd082defec433b1ca7db77c2151f32/libs/common-utils/src/resolveENSContentHash.ts#L66

 
In summary, because of 2 and 3 I would think you can't cache the result of that function

Reply:
🤦‍♂️ Sorry, I didn't get this at the beginning.
Very good note, thank you!

Reply:
Added caching for 6 hours:
https://github.com/cowprotocol/cowswap/pull/3152/commits/33cc7d85d64dd7d8a633ec656aa4dbc4585849a8